### DISCUSSION TITLE
TagCloseButton and context

### DISCUSSION THREAD

**Discussion by AlonGvili** [chakra-ui/chakra-ui#4129] [General]
Why tag component not implemented like the modal with context, so the close tag button work like the close modal button?

the close modal button uses the modal context to close the modal,
but for the tag, I need to implement this myself

I think if you offer a close button for a specific component you should implement also the actions for that component.
just a thought

**Comment by with-heart** [comment]
I'm all about ease-of-use, but I'm not sure how this would work. 

`CloseButton` works well for `Modal` and others because there is a single definite outcome (closing the modal) which comes from a single source (the `onClose` prop of the wrapping `Modal`).

What would `TagCloseButton` do exactly? 

imo since the user defines how each `Tag` is rendered, it makes sense for the responsibility to be on the user to define how it is "closed".

**Reply by chr-ge** [reply]
I agree. Often there is more going on than simply closing/hiding the tag when the close button is pressed.

**Reply by AlonGvili** [reply]
I understand, maybe combine both ideas, let say if the user supply onClose callback, use the user onClose, but if not just hide the tag ?
